src/handlers/transfers/handler.js

```diff
@@ -198,7 +198,9 @@ const prepare = async (error, messages) => {
         Logger.isInfoEnabled && Logger.info(Util.breadcrumb(location, `positionTopic1--${actionLetter}7`))
         functionality = TransferEventType.POSITION
         const eventDetail = { functionality, action }
-        await Kafka.proceed(Config.KAFKA_CONFIG, params, { consumerCommit, eventDetail, toDestination })
+        // Key position prepare message with payer account id
+        const payerAccount = await Participant.getAccountByNameAndCurrency(payload.payerFsp, payload.amount.currency, Enum.Accounts.LedgerAccountType.POSITION)
```

Comment:
For this particular case here(happy path), if we want to save a DB query, the participant account id is already fetched in [saveTransferPrepared](https://github.com/mojaloop/central-ledger/blob/c8e0d57776bfcd055df7fe981d88f8e2ccce258c/src/models/transfer/facade.js#L414) of facade. You can pass it down to access here. But as long as the query is cached, I don't think its a must. @mdebarros What do you suggest?

Reply:
Lets create a story for this enhancement.

This in practice shouldn't be an issue as long as caching is enabled, but thats not too say we shouldn't make such improvements going forward. I also think it would be good to address it as a dedicated story as there may be other such opportunities to reduce "potential" duplicate queries.

Reply:
I've created a story to optimize duplicated queries for participant currency id with a clause to investigate other instances of duplicated queries (still updating the story, just a placeholder) https://github.com/mojaloop/project/issues/3515